src/compute/src/render/join/delta_join.rs

```diff
@@ -299,6 +305,95 @@ where
     }
 }
 
+/// Dispatches half-join construction according to arrangement type specialization.
+fn dispatch_halfjoin_local<G, CF>(
+    updates: Collection<G, (Row, G::Timestamp), Diff>,
+    trace: SpecializedArrangement<G>,
+    prev_key: Vec<MirScalarExpr>,
+    prev_thinning: Vec<usize>,
+    comparison: CF,
+    closure: JoinClosure,
+    shutdown_token: ShutdownToken,
+) -> (
+    Collection<G, (Row, G::Timestamp), Diff>,
+    Collection<G, DataflowError, Diff>,
+)
+where
+    G: Scope,
+    G::Timestamp: crate::render::RenderTimestamp,
+    CF: Fn(&G::Timestamp, &G::Timestamp) -> bool + 'static,
+{
+    match trace {
+        SpecializedArrangement::Bits32Row(key_types, inner) => build_halfjoin(
+            updates,
+            inner,
+            &key_types,
+            &Vec::default(),
```

Comment:
```suggestion
            &[],
```

Nit: Pass empty slices instead of empty vectors. Requires a change to the function.

Reply:
Done!